python/across_api/across/resolve.py

```diff
@@ -0,0 +1,128 @@
+
+    Parameters
+    ----------
+    ztf_id
+        ZTF name of source
+
+    Returns
+    -------
+        RA, Dec in ICRS decimal degrees, or None, None if not found
+
+    FIXME: Replace with antares-client module call in future, once confluent-kafka-python issues are resolved.
+    """
+    search_query = json.dumps(
+        {"query": {"bool": {"filter": {"term": {"properties.ztf_object_id": ztf_id}}}}}
+    )
+
+    params = {
+        "sort": "-properties.newest_alert_observation_time",
+        "elasticsearch_query[locus_listing]": search_query,
+    }
+    r = requests.get(ANTARES_URL, params=params)
+
+    if r.status_code == 200:
+        antares_data = json.loads(r.text)
+        ra = antares_data["data"][0]["attributes"]["ra"]
+        dec = antares_data["data"][0]["attributes"]["dec"]
+        return ra, dec
+    else:
+        return None, None
+
+
+class Resolve(ACROSSAPIBase):
+    """
+    Resolve class for resolving astronomical object names.
+
+    Parameters:
+    -----------
+    name
+        The name of the astronomical object to resolve.
+
+    Attributes:
+    -----------
+    ra
+        The right ascension of the resolved object.
+    dec
+        The declination of the resolved object.
+    name
+        The name of the astronomical object.
+    resolver
+        The resolver used for resolving the object.
+
+    Methods:
+    --------
+    get():
+        Retrieves the resolved object information.
+    """
+
+    api_name: str = "Resolve"
+    _schema = ResolveSchema
+    _get_schema = ResolveGetSchema
+
+    # Type hints
+    ra: Optional[float]
+    dec: Optional[float]
+    name: str
+    resolver: Optional[str]
+
+    def __init__(self, name: str):
+        # Class specific values
+        self.ra = None
+        self.dec = None
+        self.name = name
+        self.resolver = None
+        if self.validate_get():
+            self.get()
+
+    def get(self) -> bool:
+        """
+        Retrieves the RA and Dec coordinates for a given name.
+
+        Returns
+        -------
+            True if the name is successfully resolved, False otherwise.
+        """
+        # Make sure the required parameters are given in the correct format
+        if not self.validate_get():
+            return False
+
+        """Do a name search"""
+        # Check against the ANTARES broker
+        if "ZTF" in self.name:
+            ra, dec = antares_radec(self.name)
+            if ra is not None:
+                self.ra, self.dec = ra, dec
+                self.resolver = "ANTARES"
+                return True
+
+        # Check using the CDS resolver
+        try:
+            skycoord = SkyCoord.from_name(self.name)
+            self.ra, self.dec = skycoord.ra.deg, skycoord.dec.deg
+            self.resolver = "CDS"
+            return True
+        except NameResolveError:
+            pass
+
+        # If no resolution occurred, report this to the user
+        self.resolver = "Could not resolve name."
```

Comment:
How about just returning no resolver, or None? I'm always hesitant to signal errors "in band" by using values that are allowed as normal return types --- in this case a string.

Reply:
I'm even less of a fan of code failing silently. We either need to report an error code or add some sort of passing of errors/warnings. The latter did exist before I removed the Job stuff.

Reply:
I think that returning None for ra, dec, and resolver is a pretty clear indication that nothing was found. Especially if we document that.

Reply:
3c0a8ce